[Bug]: aws_lightsail_instance, `name` verification always fails.

### Terraform Core Version

1.5.7

### AWS Provider Version

5.16.1

### Affected Resource(s)

* aws_lightsail_instance 

### Expected Behavior

* Successful verification.

### Actual Behavior

* Validation of "name" fails.

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

```hcl
provider "aws" {
  region = "eu-west-1"
}

resource "aws_lightsail_instance" "promo_site" {
  name              = "promosite"
  availability_zone = "eu-west-1a"
  blueprint_id      = "joomla_3_6_5"
  bundle_id         = "nano_1_0"
  key_pair_name     = "some_key"
}
```

### Steps to Reproduce

```bash
$ terraform init
$ TF_LOG=DEBUG terraform validate
```



### Debug Output

```
2023-09-10T03:57:02.068Z [INFO]  Terraform version: 1.5.7
2023-09-10T03:57:02.068Z [DEBUG] using github.com/hashicorp/go-tfe v1.26.0
2023-09-10T03:57:02.068Z [DEBUG] using github.com/hashicorp/hcl/v2 v2.16.2
2023-09-10T03:57:02.068Z [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.0
2023-09-10T03:57:02.068Z [DEBUG] using github.com/zclconf/go-cty v1.12.2
2023-09-10T03:57:02.068Z [INFO]  Go runtime version: go1.20.7
2023-09-10T03:57:02.068Z [INFO]  CLI args: []string{"/root/.tfenv/versions/1.5.7/terraform", "validate"}
2023-09-10T03:57:02.068Z [DEBUG] Attempting to open CLI config file: /root/.terraformrc
2023-09-10T03:57:02.068Z [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2023-09-10T03:57:02.068Z [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2023-09-10T03:57:02.068Z [DEBUG] ignoring non-existing provider search directory /root/.terraform.d/plugins
2023-09-10T03:57:02.068Z [DEBUG] ignoring non-existing provider search directory /root/.local/share/terraform/plugins
2023-09-10T03:57:02.068Z [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2023-09-10T03:57:02.068Z [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2023-09-10T03:57:02.068Z [INFO]  CLI command args: []string{"validate"}
2023-09-10T03:57:02.811Z [DEBUG] checking for provisioner in "."
2023-09-10T03:57:02.811Z [DEBUG] checking for provisioner in "/root/.tfenv/versions/1.5.7"
2023-09-10T03:57:02.811Z [DEBUG] Building and walking validate graph
2023-09-10T03:57:02.811Z [DEBUG] ProviderTransformer: "aws_lightsail_instance.promo_site" (*terraform.NodeValidatableResource) needs provider["registry.terraform.io/hashicorp/aws"]
2023-09-10T03:57:02.811Z [DEBUG] created provider logger: level=debug
2023-09-10T03:57:02.811Z [INFO]  provider: configuring client automatic mTLS
2023-09-10T03:57:02.827Z [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.16.0/linux_amd64/terraform-provider-aws_v5.16.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/aws/5.16.0/linux_amd64/terraform-provider-aws_v5.16.0_x5]
2023-09-10T03:57:02.828Z [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.16.0/linux_amd64/terraform-provider-aws_v5.16.0_x5 pid=112288
2023-09-10T03:57:02.828Z [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.16.0/linux_amd64/terraform-provider-aws_v5.16.0_x5
2023-09-10T03:57:02.908Z [INFO]  provider.terraform-provider-aws_v5.16.0_x5: configuring server automatic mTLS: timestamp=2023-09-10T03:57:02.908Z
2023-09-10T03:57:02.926Z [DEBUG] provider: using plugin: version=5
2023-09-10T03:57:02.926Z [DEBUG] provider.terraform-provider-aws_v5.16.0_x5: plugin address: address=/tmp/plugin3474112601 network=unix timestamp=2023-09-10T03:57:02.925Z
2023-09-10T03:57:03.534Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-09-10T03:57:03.543Z [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.16.0/linux_amd64/terraform-provider-aws_v5.16.0_x5 pid=112288
2023-09-10T03:57:03.543Z [DEBUG] provider: plugin exited
2023-09-10T03:57:03.543Z [DEBUG] ReferenceTransformer: "aws_lightsail_instance.promo_site" references: []
2023-09-10T03:57:03.543Z [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/aws\"]" references: []
2023-09-10T03:57:03.544Z [DEBUG] Starting graph walk: walkValidate
2023-09-10T03:57:03.544Z [DEBUG] created provider logger: level=debug
2023-09-10T03:57:03.544Z [INFO]  provider: configuring client automatic mTLS
2023-09-10T03:57:03.550Z [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.16.0/linux_amd64/terraform-provider-aws_v5.16.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/aws/5.16.0/linux_amd64/terraform-provider-aws_v5.16.0_x5]
2023-09-10T03:57:03.550Z [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.16.0/linux_amd64/terraform-provider-aws_v5.16.0_x5 pid=112312
2023-09-10T03:57:03.550Z [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.16.0/linux_amd64/terraform-provider-aws_v5.16.0_x5
2023-09-10T03:57:03.629Z [INFO]  provider.terraform-provider-aws_v5.16.0_x5: configuring server automatic mTLS: timestamp=2023-09-10T03:57:03.628Z
2023-09-10T03:57:03.645Z [DEBUG] provider: using plugin: version=5
2023-09-10T03:57:03.645Z [DEBUG] provider.terraform-provider-aws_v5.16.0_x5: plugin address: address=/tmp/plugin1207799467 network=unix timestamp=2023-09-10T03:57:03.645Z
2023-09-10T03:57:04.209Z [ERROR] provider.terraform-provider-aws_v5.16.0_x5: Response contains error diagnostic: @module=sdk.proto diagnostic_detail= diagnostic_severity=ERROR tf_req_id=30b55aff-1b74-33ea-d197-9c142e703e16 tf_resource_type=aws_lightsail_instance tf_rpc=ValidateResourceTypeConfig @caller=github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/diag/diagnostics.go:58 tf_proto_version=5.4 diagnostic_attribute=AttributeName("name") diagnostic_summary="invalid value for name (must contain only alphanumeric characters, underscores, hyphens, and dots)" tf_provider_addr=registry.terraform.io/hashicorp/aws timestamp=2023-09-10T03:57:04.209Z
2023-09-10T03:57:04.209Z [ERROR] vertex "aws_lightsail_instance.promo_site" error: invalid value for name (must contain only alphanumeric characters, underscores, hyphens, and dots)
╷
│ Error: invalid value for name (must contain only alphanumeric characters, underscores, hyphens, and dots)
│ 
│   with aws_lightsail_instance.promo_site,
│   on ai1.tf line 15, in resource "aws_lightsail_instance" "promo_site":
│   15:   name              = "promosite"
│ 
╵
2023-09-10T03:57:04.210Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-09-10T03:57:04.221Z [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.16.0/linux_amd64/terraform-provider-aws_v5.16.0_x5 pid=112312
2023-09-10T03:57:04.221Z [DEBUG] provider: plugin exited
```

### Panic Output

_No response_

### Important Factoids

Downgrade the "provider" to 5.15.0 will result in successful validation.

hoge.tf

```hcl
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "5.15.0"
    }
  }
}

provider "aws" {
  region = "eu-west-1"
}

resource "aws_lightsail_instance" "promo_site" {
  name              = "promosite"
  availability_zone = "eu-west-1a"
  blueprint_id      = "joomla_3_6_5"
  bundle_id         = "nano_1_0"
  key_pair_name     = "some_key"
}

```

operation

```bash
$ rm -rf .terraform .terraform.lock.hcl
$ terraform init
$ TF_LOG=DEBUG terraform validate
```

output

```
2023-09-10T03:58:38.419Z [INFO]  Terraform version: 1.5.7
2023-09-10T03:58:38.419Z [DEBUG] using github.com/hashicorp/go-tfe v1.26.0
2023-09-10T03:58:38.419Z [DEBUG] using github.com/hashicorp/hcl/v2 v2.16.2
2023-09-10T03:58:38.419Z [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.0
2023-09-10T03:58:38.419Z [DEBUG] using github.com/zclconf/go-cty v1.12.2
2023-09-10T03:58:38.419Z [INFO]  Go runtime version: go1.20.7
2023-09-10T03:58:38.419Z [INFO]  CLI args: []string{"/root/.tfenv/versions/1.5.7/terraform", "init"}
2023-09-10T03:58:38.419Z [DEBUG] Attempting to open CLI config file: /root/.terraformrc
2023-09-10T03:58:38.419Z [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2023-09-10T03:58:38.419Z [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2023-09-10T03:58:38.419Z [DEBUG] ignoring non-existing provider search directory /root/.terraform.d/plugins
2023-09-10T03:58:38.419Z [DEBUG] ignoring non-existing provider search directory /root/.local/share/terraform/plugins
2023-09-10T03:58:38.419Z [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2023-09-10T03:58:38.419Z [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2023-09-10T03:58:38.420Z [INFO]  CLI command args: []string{"init"}

Initializing the backend...
2023-09-10T03:58:38.421Z [DEBUG] New state was assigned lineage "4b6b82c5-4d2f-0082-9182-1dc85a3b9cf5"
2023-09-10T03:58:38.421Z [DEBUG] checking for provisioner in "."
2023-09-10T03:58:38.421Z [DEBUG] checking for provisioner in "/root/.tfenv/versions/1.5.7"

Initializing provider plugins...
- Finding hashicorp/aws versions matching "5.15.0"...
2023-09-10T03:58:38.421Z [DEBUG] Service discovery for registry.terraform.io at https://registry.terraform.io/.well-known/terraform.json
2023-09-10T03:58:38.688Z [DEBUG] GET https://registry.terraform.io/v1/providers/hashicorp/aws/versions
2023-09-10T03:58:38.882Z [DEBUG] GET https://registry.terraform.io/v1/providers/hashicorp/aws/5.15.0/download/linux/amd64
2023-09-10T03:58:39.085Z [DEBUG] GET https://releases.hashicorp.com/terraform-provider-aws/5.15.0/terraform-provider-aws_5.15.0_SHA256SUMS
2023-09-10T03:58:39.360Z [DEBUG] GET https://releases.hashicorp.com/terraform-provider-aws/5.15.0/terraform-provider-aws_5.15.0_SHA256SUMS.72D7468F.sig
- Installing hashicorp/aws v5.15.0...
2023-09-10T03:58:43.503Z [DEBUG] Provider signed by 34365D9472D7468F HashiCorp Security (hashicorp.com/security) <security@hashicorp.com>
- Installed hashicorp/aws v5.15.0 (signed by HashiCorp)

Terraform has created a lock file .terraform.lock.hcl to record the provider
selections it made above. Include this file in your version control repository
so that Terraform can guarantee to make the same selections by default when
you run "terraform init" in the future.

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```


### References

_No response_

### Would you like to implement a fix?

No

--- a/.changelog/33405.txt
+++ b/.changelog/33405.txt
@@ -1,0 +1,39 @@
+```release-note:bug
+resource/aws_lightsail_certificate: Fix validation of `name`
+```
+
+```release-note:bug
+resource/aws_lightsail_database: Fix validation of `name`
+```
+
+```release-note:bug
+resource/aws_lightsail_disk: Fix validation of `name`
+```
+
+```release-note:bug
+resource/aws_lightsail_instance: Fix validation of `name`
+```
+
+```release-note:bug
+resource/aws_lightsail_lb: Fix validation of `lb_name`
+```
+
+```release-note:bug
+resource/aws_lightsail_lb_attachment: Fix validation of `lb_name`
+```
+
+```release-note:bug
+resource/aws_lightsail_lb_certificate: Fix validation of `lb_name`
+```
+
+```release-note:bug
+resource/aws_lightsail_lb_certificate_attachment: Fix validation of `lb_name`
+```
+
+```release-note:bug
+resource/aws_lightsail_lb_https_redirection_policy: Fix validation of `lb_name`
+```
+
+```release-note:bug
+resource/aws_lightsail_lb_stickiness_policy: Fix validation of `lb_name`
+```

--- a/internal/service/lightsail/certificate.go
+++ b/internal/service/lightsail/certificate.go
@@ -90,7 +90,7 @@ func ResourceCertificate() *schema.Resource {
 				ValidateFunc: validation.All(
 					validation.StringLenBetween(2, 255),
 					validation.StringMatch(regexache.MustCompile(`^[A-Za-z]`), "must begin with an alphabetic character"),
-					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[_.^-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
+					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[^_.-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
 				),
 			},
 			"subject_alternative_names": {

--- a/internal/service/lightsail/database.go
+++ b/internal/service/lightsail/database.go
@@ -165,7 +165,7 @@ func ResourceDatabase() *schema.Resource {
 				ForceNew: true,
 				ValidateFunc: validation.All(
 					validation.StringLenBetween(2, 255),
-					validation.StringMatch(regexache.MustCompile(`^[_.^-]+[0-9A-Za-z-]+[_.^-]$`), "Must contain from 2 to 255 alphanumeric characters, or hyphens. The first and last character must be a letter or number"),
+					validation.StringMatch(regexache.MustCompile(`^[^_.-]+[0-9A-Za-z-]+[^_.-]$`), "Must contain from 2 to 255 alphanumeric characters, or hyphens. The first and last character must be a letter or number"),
 				),
 			},
 			"secondary_availability_zone": {

--- a/internal/service/lightsail/disk.go
+++ b/internal/service/lightsail/disk.go
@@ -57,7 +57,7 @@ func ResourceDisk() *schema.Resource {
 				ValidateFunc: validation.All(
 					validation.StringLenBetween(2, 255),
 					validation.StringMatch(regexache.MustCompile(`^[A-Za-z]`), "must begin with an alphabetic character"),
-					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[_.^-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
+					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[^_.-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
 				),
 			},
 			"size_in_gb": {

--- a/internal/service/lightsail/instance.go
+++ b/internal/service/lightsail/instance.go
@@ -71,7 +71,7 @@ func ResourceInstance() *schema.Resource {
 				ValidateFunc: validation.All(
 					validation.StringLenBetween(2, 255),
 					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z]`), "must begin with an alphanumeric character"),
-					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[_.^-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
+					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[^_.-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
 				),
 			},
 			"availability_zone": {

--- a/internal/service/lightsail/lb.go
+++ b/internal/service/lightsail/lb.go
@@ -85,7 +85,7 @@ func ResourceLoadBalancer() *schema.Resource {
 				ValidateFunc: validation.All(
 					validation.StringLenBetween(2, 255),
 					validation.StringMatch(regexache.MustCompile(`^[A-Za-z]`), "must begin with an alphabetic character"),
-					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[_.^-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
+					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[^_.-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
 				),
 			},
 			"support_code": {

--- a/internal/service/lightsail/lb_attachment.go
+++ b/internal/service/lightsail/lb_attachment.go
@@ -41,7 +41,7 @@ func ResourceLoadBalancerAttachment() *schema.Resource {
 				ValidateFunc: validation.All(
 					validation.StringLenBetween(2, 255),
 					validation.StringMatch(regexache.MustCompile(`^[A-Za-z]`), "must begin with an alphabetic character"),
-					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[_.^-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
+					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[^_.-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
 				),
 			},
 			"instance_name": {

--- a/internal/service/lightsail/lb_certificate.go
+++ b/internal/service/lightsail/lb_certificate.go
@@ -87,7 +87,7 @@ func ResourceLoadBalancerCertificate() *schema.Resource {
 				ValidateFunc: validation.All(
 					validation.StringLenBetween(2, 255),
 					validation.StringMatch(regexache.MustCompile(`^[A-Za-z]`), "must begin with an alphabetic character"),
-					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[_.^-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
+					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[^_.-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
 				),
 			},
 			"name": {
@@ -97,7 +97,7 @@ func ResourceLoadBalancerCertificate() *schema.Resource {
 				ValidateFunc: validation.All(
 					validation.StringLenBetween(2, 255),
 					validation.StringMatch(regexache.MustCompile(`^[A-Za-z]`), "must begin with an alphabetic character"),
-					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[_.^-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
+					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[^_.-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
 				),
 			},
 			"subject_alternative_names": {

--- a/internal/service/lightsail/lb_certificate_attachment.go
+++ b/internal/service/lightsail/lb_certificate_attachment.go
@@ -42,7 +42,7 @@ func ResourceLoadBalancerCertificateAttachment() *schema.Resource {
 				ValidateFunc: validation.All(
 					validation.StringLenBetween(2, 255),
 					validation.StringMatch(regexache.MustCompile(`^[A-Za-z]`), "must begin with an alphabetic character"),
-					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[_.^-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
+					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[^_.-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
 				),
 			},
 			"certificate_name": {

--- a/internal/service/lightsail/lb_https_redirection_policy.go
+++ b/internal/service/lightsail/lb_https_redirection_policy.go
@@ -45,7 +45,7 @@ func ResourceLoadBalancerHTTPSRedirectionPolicy() *schema.Resource {
 				ValidateFunc: validation.All(
 					validation.StringLenBetween(2, 255),
 					validation.StringMatch(regexache.MustCompile(`^[A-Za-z]`), "must begin with an alphabetic character"),
-					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[_.^-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
+					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[^_.-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
 				),
 			},
 		},

--- a/internal/service/lightsail/lb_stickiness_policy.go
+++ b/internal/service/lightsail/lb_stickiness_policy.go
@@ -50,7 +50,7 @@ func ResourceLoadBalancerStickinessPolicy() *schema.Resource {
 				ValidateFunc: validation.All(
 					validation.StringLenBetween(2, 255),
 					validation.StringMatch(regexache.MustCompile(`^[A-Za-z]`), "must begin with an alphabetic character"),
-					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[_.^-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
+					validation.StringMatch(regexache.MustCompile(`^[0-9A-Za-z_.-]+[^_.-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
 				),
 			},
 		},

